mlir/utils/jenkins/Jenkinsfile

```diff
@@ -1057,5 +1052,68 @@ pipeline {
+            matrix {
+                axes {
+                    axis {
+                        name 'CPATH'
+                        values 'mfma'           // 'navi3x' or hypothetical 'wmma' if needed
+                    }
+                }
+                agent {
+                    docker {
+                        image dockerImage()
+                        args dockerArgs()
+                        label getLabelFromCodepath("${CPATH}")
+                        alwaysPull true
+                    }
+                }
+                environment {
+                    PATH="/opt/rocm/llvm/bin:$PATH"
+                    HOME="${WORKSPACE}"
+                    // Note the %m to avoid issues with threads and dynamic libraries.
+                    LLVM_PROFILE_FILE="${WORKSPACE}/build/%m-%p.profraw"
+                }
+                stages {
+                    stage ("body") {
+                        steps {
+                            // Build with profiling on, and just code-generation tests.
+                            sh 'rm -f build/CMakeCache.txt'
+                            sh 'rm -f build/*.profraw'
+                            buildProject('check-rocmlir-build-only',
+                                         '-DBUILD_FAT_LIBROCKCOMPILER=ON -DCMAKE_BUILD_TYPE=debug -DLLVM_BUILD_INSTRUMENTED_COVERAGE=ON')
+                            dir ('build') {
+                                // Run tests.
+                                sh 'ninja check-rocmlir'
+                                // Profile processing.
+                                sh '/opt/rocm/llvm/bin/llvm-profdata merge -sparse ./*.profraw -o ./coverage.profdata'
+                                sh "/opt/rocm/llvm/bin/llvm-cov report --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project > ./coverage_${CPATH}.report"
+                                sh "cat ./coverage_${CPATH}.report"
+                                sh "/opt/rocm/llvm/bin/llvm-cov export --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project --format=lcov --compilation-dir ${WORKSPACE} > ./coverage_${CPATH}.lcov"
+                                sh "/opt/rocm/llvm/bin/llvm-cov show --object ./bin/rocmlir-opt --object ./bin/rocmlir-driver --object ./bin/rocmlir-gen --instr-profile ./coverage.profdata --ignore-filename-regex=external/llvm-project -Xdemangler=/opt/rocm/llvm/bin/llvm-cxxfilt --format=html > ./coverage_${CPATH}.html"
+                                // Upload to codecov.
+                                withCredentials([string(credentialsId: 'codecov-token-rocmlir', variable: 'CODECOV_TOKEN')]) {
+                                    sh '''
+                                       curl -Os https://uploader.codecov.io/latest/linux/codecov && chmod +x ./codecov
```

Comment:
Upload to codecov.io as they recommend.  Use the "-U" proxy option for lockhart, which otherwise can't reach it.